### PR TITLE
Fix tcp dial overhead

### PR DESCRIPTION
**Pull request by shyamjesal**
- removed TCP dial overhead b/w dQP and DST from critical path
- fix benchmark test artifact path
- - removed TCP dial overhead b/w sQP and SRC

### CODE REVIEW

File: tests/integration_test.go [hunk 1]
```diff
@@ -173,7 +173,7 @@ func TestSdk_InvokeWithXDT(t *testing.T) {
 	start := time.Now()
 	log.Infof("starting integ test")
 	url := config.ProxyHostname + config.ProxyPort
-	if err := sdk.InvokeWithXDT(url, preparePayload(), config.SQPServerHostname+config.SQPServerPort, chunkSizeInBytes); err != nil {
+	if err := sdk.InvokeWithXDT(url, preparePayload(), config.SQPServerHostname+config.SQPServerPort, chunkSizeInBytes, config.Routing, config.RPCTimeoutDuration); err != nil {
```
Comment (ustiugov) [inline tests/integration_test.go]: a very long line

File: sdk/golang/source.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ func splitPayload(xdtPayload *utils.Payload) (string, []byte) {
 }
 
 // InvokeWithXDT invokes the RPC call with XDT
-func InvokeWithXDT(URL string, xdtPayload utils.Payload, sQPAddr string, chunkSizeInBytes int) error {
+func InvokeWithXDT(URL string, xdtPayload utils.Payload, sQPAddr string, chunkSizeInBytes int, routing string, RPCTimeout int) error {
```
Comment (ustiugov) [inline sdk/golang/source.go]: a very long line

**Review by ustiugov — CHANGES_REQUESTED** [review]
- [x] splint long lines into many, lots of super long function definitions. 100 characters should be MAX
- [x] get rid of the init-time parameters inside RPCs

**Comment by shyamjesal** [comment]
Both the TCP dial overheads are removed. I think the sQP-Src overhead is still in the critical path. Can you think of a way to solve this? How about making the connection in a separate go-routine?

File: sdk/golang/source.go [hunk 3]
```diff
@@ -40,6 +40,18 @@ import (
 	"google.golang.org/grpc"
 )
```
Comment (ustiugov) [inline sdk/golang/source.go:42]: I suggest a minor refactoring: define a type, called XDT client. Then InitXDT would become a part of NewXDTClient call and Invoke(WithXDT postfix can be omitted) as well as PushData become the methods of that type. 

This way you don't need to pass the client as an argument

**Review by ustiugov — CHANGES_REQUESTED** [review]

**Review by ustiugov — APPROVED** [review]

**Pull request merged**